src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -373,7 +389,7 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
       .result();
 
     // When the zone is stable initially, and when the option list changes...
-    return RxChain.from(merge(firstStable, optionChanges))
+    return RxChain.from(merge(this._nextStable, optionChanges))
```

Comment:
Considering that this is the only place left that uses `_nextStable`, consider inlining it again?

Reply:
Unsure about this suggestion. By doing it my way, we can avoid making NgZone a class member. I also imagine that it's slightly more performant to not reconstruct Observables everywhere (this optimization can be done in several places throughout the code base).

The benefits that I listed are not significant, so if you feel strongly about it, I will revert.

Reply:
While this way the `NgZone` isn't a class member, `NgZone.onStable` still is. By having `NgZone` be the class member, it's easier to access if we need to use it anywhere else. Also the performance difference should be negligible. Also one more gotcha about the `onStable`, can you switch it to `onStable.asObservable()`? We've had some memory leaks in the past because we weren't converting it into an observable.

Reply:
> By having NgZone be the class member, it's easier to access if we need to use it anywhere else. 

I consider this a negative since I tend to believe in limiting the scope of class member to what the class needs. 

`onStable.asObservable()` should not be necessary with the `first` call since according to [What are operators?](http://reactivex.io/rxjs/manual/overview.html#what-are-operators-):

> Operators are methods on the Observable type, such as .map(...), .filter(...), .merge(...), etc. When called, they do not change the existing Observable instance. Instead, they return a new Observable, whose subscription logic is based on the first Observable.

Anyway, I do believe in having smaller diffs, though, so I've made your suggestions and reverted both changes. Now the `_subscribeToClosingActions` method is unchanged from master.

Reply:
`onStable.asObservable()` definitely seems like it shouldn't, but it ends up causing any references inside the callback to be retained in memory after the component is destroyed. I'm not completely sure why that is the case, but I think it has to do with the `onStable` being an `EventEmitter`, not an `Observable` and the `NgZone` being a singleton.

Reply:
That makes sense to me if you were subscribing directly to onStable, but it happened even when you applied an operator before subscribing?

Anyway, thanks for the heads up. It seems like a practice that I should adopt to be safe in any case.

Reply:
Yeah it was a little hard to track down, but we were doing in a lot of places. See https://github.com/angular/material2/pull/6918